File: A/sol.cpp
//
// Author: Moncef Mhasni
// Main solution
//
#include <bits/stdc++.h>
#define ll long long
using namespace std;

ll solve(ll n){
    if(n<=2) return 1;
    ll res = n;
    for(ll i = 2; i <= n ; i++){
        if(n%i == 0){
            res -= res/i;
            while(n%i == 0) n/=i;
        }
    }
    return res;
}

int main(){
    int tc;
    scanf("%d",&tc);
    while(tc--){
        ll n; scanf("%lld",&n);
        printf("%lld\n",solve(n));
    }
}

File: A/sol.py
# author: Azuz Eraoui

def solve(n):
    res = n
    i = 2

    while i * i <= n:
        if n % i == 0:
            while n % i == 0:
                n = n // i
            res = res - res // i

        i = i + 1
    if n > 1 :
        res -= res // n
    return res

t = int(input())
for ttt in range(t):
    print(solve(int(input())))
